tools/ReleaseNotes/release_editor.py

```diff
@@ -7,120 +7,140 @@
+def create_release_note_directory_path(amalgamate_line: str, amalgamate_directive: str, release_notes_root: pathlib.Path) -> pathlib.Path:
+    release_note_dir = amalgamate_line.replace(amalgamate_directive, "").strip()
+    return release_notes_root / release_note_dir
 
 
-def addReleaseNotesToMain(pathDirectory):
+def add_release_notes_to_main_pages(release_notes_root: pathlib.Path, git_token: str) -> None:
     # iterates through files in a directory
-    for file in pathDirectory.glob("*.rst"):
-        with open(file) as fileToEdit:
+    for file in release_notes_root.glob("*.rst"):
+        with open(file) as f:
             # iterate through each line in the upper level release note file e.g. diffraction.rst
-            for line in fileToEdit:
+            for line in f:
                 # finds the amalgamate directive to replace
                 if line.startswith(DIRECTIVE):
-                    fileName = createFileName(line, pathDirectory, DIRECTIVE)
-                    releaseNotes = collateNotes(fileName)
-                    originalFile = mainDirectory / file
-                    updateFile(originalFile, releaseNotes, line)
+                    release_note_sub_dir = create_release_note_directory_path(line, DIRECTIVE, release_notes_root)
+                    release_notes_text = collate_release_notes(release_note_sub_dir, git_token)
+                    update_file(release_notes_root / file, release_notes_text, line)
 
 
-def getReleaseNoteDirectories(path, directoryList):
-    # return nothing if path is a file
-    if os.path.isfile(path):
-        return ()
+def append_release_note_directories(path: pathlib.Path, directory_list: List[pathlib.Path]):
+    if not os.path.isdir(path):
+        return
 
     # add dir to directorylist if it contains .rst files
     if len([f for f in os.listdir(path) if f.endswith(".rst")]) > 0:
-        path = pathlib.Path(path)
-        directoryList.append(path)
+        directory_list.append(path)
 
     for d in os.listdir(path):
-        new_path = os.path.join(path, d)
-        if os.path.isdir(new_path):
-            getReleaseNoteDirectories(new_path, directoryList)
+        append_release_note_directories(path / d, directory_list)
+
+
+def gather_release_note_directories(path: pathlib.Path) -> List[pathlib.Path]:
+    release_note_directories = []
+    append_release_note_directories(path, release_note_directories)
+    release_note_directories.pop(0)
+    return release_note_directories
 
-    return directoryList
 
+def get_pr_number_and_link(release_note_path: pathlib.Path, git_token: str) -> Tuple[str, str]:
+    commit_hashes = get_file_creation_commits(release_note_path)
+    for commit_hash in commit_hashes:
+        headers = {"Authorization": f"token {git_token}", "Accept": "application/vnd.github.groot-preview+json"}
+        url = f"https://api.github.com/repos/{GIT_REPO}/commits/{commit_hash}/pulls"
+        response = requests.get(url, headers=headers)
```

Comment:
```suggestion
        response = requests.get(url, headers=headers, timeout=10)
```
Not having a timeout breaks [this](https://docs.astral.sh/ruff/rules/request-without-timeout/) rule